jumanji/environments/routing/op/generator.py

```diff
@@ -0,0 +1,167 @@
+import abc
+import chex
+import jax
+import jax.numpy as jnp
+
+import types_, constants
+
+class Generator(abc.ABC):
+    """
+    Defines the abstract 'Generator' base class. A 'Generator' is responsible
+    for generating a problem instance when the environment is reset
+    """
+
+    def __init__(self, num_nodes: int, max_length: int):
+        """Abstract class implementing the attributes 'num_nodes' and 'max_length'.
+        
+        Args:
+            num_nodes (int): the number of nodes in the problem instance.
+            max_length (int): the maximum length of the tour
+        """
+        
+        self.num_nodes = num_nodes
+        self.max_length = max_length
+        
+      
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey) -> types_.State:
+        """Call method responsible for generating a new state.
+        Args:
+            key: jax random key in case stochasticity is used in the instance generation process
+            
+        Returns:
+            An 'OP' environment state.
+        """    
+        
+        
+class ConstantGenerator(Generator):
+    """Instance generator that generates a constant instance of the orienteering problem.
+    Given the number of nodes and maximum route length, the generator works as follows: the 
+    coordinates of the nodes (including the depot) are randomly sampled from a  uniform 
+    distribution of a unit square and the prizes are set constant such that every node has the 
+    same prize. The prize at the depot is 0. The length between the depot and the nodes  is 
+    randomly sampled from the interval (0, max_length).
+    """
+    
+    def __init__(self, num_nodes: int, max_length: int):
```

Comment:
Not useful here. You would have the same behavior by just not rewriting this function.